### DISCUSSION TITLE
In multiplayer, firing with the grenade launcher before being stunned will cause the 40mm grenade to drop.

### DISCUSSION THREAD

**Discussion by zhu-rengong** [FakeFishGames/Barotrauma#14062] [Bug reports]
### Disclaimers

- [X] I have searched the issue tracker to check if the issue has already been reported.
- [ ] My issue happened while using mods.

### What happened?

As the title said.
Test video:

https://github.com/FakeFishGames/Barotrauma/assets/73634685/5ad79426-559e-480a-9473-2f1f517a1870

### Reproduction steps

1. Host a server and start a round with a submarine that's easy to reproduce like the above video content
2. Spawn a grenade launcher and load a 40mm grenade
3. Fire immediately before inflicting a slight stun on yourself, such as falling from a height

### Bug prevalence

Happens every now and then

### Single player or multiplayer?

Multiplayer hosted from the in-game menu (= using a listen server)

### Version

v1.4.6.0 (Blood in the Water Update, hotfix 2)

### Which operating system did you encounter this bug on?

Windows

**Comment by A2ur3** [comment]
shotgun shell and revolver round also have the same issue

**Comment by A2ur3** [comment]
https://github.com/user-attachments/assets/93ff646f-ef64-4512-b499-7187089ec568

can be easily reproduced
sometimes the grenade seems explode and disappear but they didn't actually
other players can see it and put it back into container these grenades will be non interactable in your sight
